src/main/java/core/basesyntax/model/io/DataConverterImpl.java

```
@@ -0,0 +1,27 @@
+package core.basesyntax.model.io;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> lines) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+
+        for (String line : lines) {
+            String[] parts = line.split(",");
+
+            if (parts.length != 3) {
+                continue;
+            }
+
+            FruitTransaction.Operation op =
+                    FruitTransaction.Operation.fromCode(parts[0].trim());
+            String fruit = parts[1].trim();
+            int quantity = Integer.parseInt(parts[2].trim());
```

Parsing the quantity with `Integer.parseInt(parts[2].trim())` has no try/catch. If the quantity is not a valid integer this will throw NumberFormatException without context. Per checklist: don’t ignore exceptions and rethrow a RuntimeException with an informative message and cause (e.g., include the offending line). Wrap parse errors and throw an informative RuntimeException instead of letting a raw NumberFormatException propagate. Also validate that the parsed quantity is not negative: "Quantity should be an integer and not negative."